packages/celotool/src/lib/geth.ts

```diff
@@ -486,23 +496,66 @@ export const transferCeloDollars = async (
     feeCurrency: txOptions.feeCurrency,
     gatewayFeeRecipient: txOptions.gatewayFeeRecipient,
     gatewayFee: txOptions.gatewayFee,
+    nonce: txOptions.nonce,
   })
 }
 
+export const unlock = async (
+  kit: ContractKit,
+  address: string,
+  password: string,
+  unlockPeriod: number,
+  unlockNeeded: boolean
+) => {
+  if (unlockNeeded) {
+    try {
+      kit.web3.eth.personal.unlockAccount(address, password, unlockPeriod)
+    } catch (error) {
+      console.error(`Unlock account ${address} failed:`, error)
+    }
+  }
+}
+
 export const simulateClient = async (
   senderAddress: string,
   recipientAddress: string,
   txPeriodMs: number, // time between new transactions in ms
   blockscoutUrl: string,
   blockscoutMeasurePercent: number, // percent of time in range [0, 100] to measure blockscout for a tx
-  index: number
+  index: number,
+  thread: number,
+  web3Provider: string = 'http://localhost:8545'
 ) => {
   // Assume the node is accessible via localhost with senderAddress unlocked
-  const kit = newKitFromWeb3(new Web3('http://localhost:8545'))
+  const kit = newKitFromWeb3(new Web3(web3Provider))
+  const password = fetchEnv('PASSWORD')
+
+  let lastNonce: number = 0
+  let lastTx: string = ''
+  let nonce: number = 0
+  let unlockNeeded: boolean = true
+
+  const sleepTime = 5000
+  while (await kit.connection.isSyncing()) {
+    console.info(
+      `LoadTestId ${index} waiting for web3Provider to be synced. Sleeping ${sleepTime}ms`
+    )
+    await sleep(sleepTime)
+  }
   kit.defaultAccount = senderAddress
 
+  // sleep a random amount of time in the range [0, txPeriodMs/5secs) before starting so
+  // that if multiple simulations are started at the same time, they don't all
+  // submit transactions at the same time
+  const maxSleep = 5000 // Sleep 5 seg at most
+  const randomSleep = Math.random() * txPeriodMs
+  const sleepMs = randomSleep > maxSleep ? maxSleep : randomSleep
```

Comment:
Why is 5 seconds the max?  We had previously used longer periods like 60s to have lots of accounts but not so many transactions.  The way is it here also means that if you give a `txPeriodMs` considerably larger than 5 seconds then most of the time `sleepMs` will be `maxSleep` and so there will be almost no randomness involved, which defeats the purpose of using random.

If we want to cap the wait at 5 seconds, perhaps we should apply it by capping `txPeriodMs` instead of capping `randomSleep`.  But I'm not clear on why we want to cap it in the first place.

Reply:
Also, since there is also a sleep below, inside `transferFn`, I'm not clear on why this sleep involves `txPeriodMs`.  If the purpose of this sleep is just to add jitter, `txPeriodMs` shouldn't come into it.  While if it is to space out the transactions from this thread, why do we need the other sleep below?  Though I may be missing something.

Reply:
The intention for this sleep is add a jitter to the threads inside the same pod. Originally was between [0, txPeriodMs], but for a scenario we were testing (high number of pods and high number of txPeriodMs), having the maxSleep helped to reach the testing scenario faster.
I'll set as it was before that probably fits better in most scenarios.